main/settings/src/main/scala/sbt/std/InputWrapper.scala

```diff
@@ -96,17 +96,22 @@ object InputWrapper {
       val tpe = ts.tree.tpe
       if (tpe <:< c.weakTypeOf[Initialize[Task[T]]])
         InputWrapper.wrapInitTask[T](c)(ts, pos)
-      else if (tpe <:< c.weakTypeOf[Initialize[T]])
-        InputWrapper.wrapInit[T](c)(ts, pos)
+      else if (tpe <:< c.weakTypeOf[Initialize[T]]) {
+        if (c.weakTypeOf[T] <:< c.weakTypeOf[InputTask[_]]) {
+          c.abort(pos, """`value` is no longer allowed for an input task. Use `evaluated` or `toInputTask`.
+                         |See http://www.scala-sbt.org/0.13/docs/Input-Tasks.html for more details.""".stripMargin)
+        } else InputWrapper.wrapInit[T](c)(ts, pos)
+      }
       else if (tpe <:< c.weakTypeOf[Task[T]])
         InputWrapper.wrapTask[T](c)(ts, pos)
-      else if (tpe <:< c.weakTypeOf[InputTask[T]])
-        InputWrapper.wrapInputTask[T](c)(ts, pos)
```

Comment:
I think I was wrong on the initial analysis. It doesn't match during `.value` but it does get used in some other code path. Added it back 725028553c56a151ae8360dd37ed817a8efc13a0 with a better test.